Sources/CoreData/Insert/DataRequest.swift

```diff
@@ -159,4 +159,68 @@ public extension DataRequest {
 			handler(response, save)
 		}
 	}
+
+	/// Adds a handler to be called once the request has finished.
+	///
+	/// - parameter type:              The `Insertable` type that will be used in the serialization
+	/// - parameter db:                The database to work in
+	/// - parameter queue:             The queue on which the deserializer (and your completion handler) is dispatched.
+	/// - parameter jsonOptions:       JSON decoder options
+	/// - parameter jsonTransformer:   The response JSON transformer
+	///
+	/// - returns: the response
+	@available(iOS 13, *)
+	func responseInsert<T: Insertable>(
+		of type: T.Type = T.self,
+		db: DB = DB.shared,
+		jsonOptions: JSONSerialization.ReadingOptions = .allowFragments,
+		jsonTransformer: @escaping (Any) throws -> Any = { $0 },
+		contextObject: Any? = nil,
+		automaticallyCancelling: Bool = false
+	) async -> DataResponse<T, AFError> {
+		let context = db.newBackgroundContext()
+		let save: DB.SaveBlockWitCallback = { completion in
+			context.perform {
+				do {
+					try context.save()
+					completion(nil)
+				} catch {
+					completion(error)
+				}
+			}
+		}
+
+		let serializer = InsertResponseSerializer(
+			jsonOptions: jsonOptions,
+			jsonTransformer: jsonTransformer,
+			context: context,
+			type: type,
+			contextObject: contextObject
+		)
+
+		let task = serializingResponse(using: serializer, automaticallyCancelling: automaticallyCancelling)
```

Comment:
Namely this will (maybe) invoke `InsertResponseSerializer.serialize(request:response:data:error:)` on some random thread instead of the context's thread. At least with the `save` above we're wrapping it in `context.perform`.

Reply:
Both async and non-async variants will always be dispatched on a serialization queue defined on the `Session`. So the  behavior is exactly the same.